pkg/database/dbDataStore.go

```diff
@@ -0,0 +1,156 @@
+package database
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/thanhpk/randstr"
+)
+
+type DatabaseDataStore struct {
+}
+
+func (c DatabaseDataStore) GetPublicProfileNames() []string {
+	return DbListPublicProfiles()
+}
+
+func (c DatabaseDataStore) GetAllProfileNames() []string {
+	return DbListProfiles()
+}
+
+func (c DatabaseDataStore) ProfileExists(name string) bool {
+	return DbProfileExists(name)
+}
+
+func (c DatabaseDataStore) GetProfileByName(name string) Profile {
+	return *DbReadProfile(name)
+}
+
+func (c DatabaseDataStore) AddProfile(name string, sources []string, public bool, immutablePast bool) {
+	DbWriteProfile(Profile{
```

Comment:
This does not add new sources, thus failing POST /api/profiles/<profilename> call to create a new profile

Reply:
see sys-testcase-1